src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -199,6 +210,12 @@ public string PropertiesFileName
             {
                 MultiFileAnalysis = true;
             }
+
+            if (MultiFileAnalysis && cmdLineProperties.GetAllProperties().FirstOrDefault(x => WarningParameters.Contains(x.Id)) is { } property)
```

Comment:
I wouldn't do this here.
Check `AdditionalFilesService`, we need to change `HasUserSpecifiedSonarTests` to check all of these properties and then return empty collections.